pkg/sources/gitlab/gitlab.go

```diff
@@ -252,22 +256,24 @@ func (s *Source) getRepos() ([]*url.URL, []error) {
 				continue
 			}
 
-			// The repo normalization has already successfully parsed the URL at this point, so we can ignore the error.
-			u, _ := url.ParseRequestURI(repo)
-			validRepos = append(validRepos, u)
+			validRepos = append(validRepos, repo)
```

Comment:
**Question:** I'm not sure if we can trust the `repo` url directly or if we should do `url.ParseRequestURI` followed by `String()`. I would think parsing would help normalize the input, but perhaps it's redundant at this point.

Reply:
Ah. The repo URL was already normalized when `giturl.NormalizeGitlabRepo(prj)` was run on it above. That eventually calls `url.Parse` here: https://github.com/trufflesecurity/trufflehog/blob/4218c39d99b5f30153f62471c1be1c1596f0a4d4/pkg/giturl/giturl.go#L41

Though now that you mention it, maybe `giturl` should be using `url.ParseRequestURI` instead of just `url.Parse`.